jsapp/js/components/submissions/attachmentActionsDropdown.component.tsx

```diff
@@ -0,0 +1,179 @@
+  };
+
+  let attachmentTypeName = t('attachment');
+  if (props.questionType === QuestionTypeName.audio) {
+    attachmentTypeName = t('audio recording');
+  } else if (props.questionType === QuestionTypeName.video) {
+    attachmentTypeName = t('video recording');
+  } else if (props.questionType === QuestionTypeName.image) {
+    attachmentTypeName = t('image');
+  } else if (props.questionType === MetaQuestionTypeName['background-audio']) {
+    attachmentTypeName = t('background audio recording');
+  }
+
+  function confirmDelete() {
+    setIsDeletePending(true);
+
+    // TODO: replace the timeout with actual API call that will delete
+    // the attachment.
+    console.log('confirmDelete');
+    setTimeout(() => {
+      // TODO: Upon finishing we need to have the submission data being updated
+      // both here in Submission Modal and in Data Table.
+      // Validation status changing in Submission Modal works like this:
+      // 1. Does the call (both Submission Modal and Data Table listens to same call)
+      // 2. Call finishes and returns a fresh SubmissionResponse
+      // 3. Upon finishing, Submission Modal updates the submission from props
+      // 4. Upon finishing, Data Table updates the submission in the list
+      // We would need to do something similar here.
+
+      // TODO: We would need to confirm from Back-end how would the deleted
+      // attachment be marked
+      onAttachmentDeleted();
+    }, 2000);
+  }
+
+  /**
+   * Stops pending animation, closes confirmation prompt and displays
+   * a notification. Also let the parent know through prop callback.
+   */
+  function onAttachmentDeleted() {
+    setIsDeletePending(false);
+    toggleDeleteModal();
+    utils.notify(
+      t('##Attachment_type## deleted')
+      .replace('##Attachment_type##', attachmentTypeName)
+    );
+    props.onDeleted();
+  }
+
+  function requestDownloadFile() {
+    utils.downloadUrl(props.attachmentUrl);
+  }
+
+  const userCanDelete = userHasPermForSubmission(
+    'delete_submissions',
+    props.asset,
+    props.submissionData
+  );
+  const userCanChange = userHasPermForSubmission(
+    'change_submissions',
+    props.asset,
+    props.submissionData
+  );
+
+  const uniqueDropdownName = `attachment-actions-${utils.generateUuid()}`;
+
+  return (
+    <>
+      <KoboDropdown
+        name={uniqueDropdownName}
+        placement='down-right'
+        hideOnMenuClick
+        triggerContent={
+          <Button
+            type='bare'
+            color='dark-blue'
+            size='s'
+            startIcon='more'
+          />
+        }
+        menuContent={
+          <bem.KoboSelect__menu>
+            <bem.KoboSelect__option onClick={requestDownloadFile}>
+              <Icon name='download' />
+              <label>{t('Download')}</label>
+            </bem.KoboSelect__option>
+
+            <bem.KoboSelect__option
+              onClick={toggleDeleteModal}
+              disabled={!userCanChange}
```

Comment:
Is there a reason we are allowing users without delete permission to click this button, but not to actually delete the attachment when they get to the modal?

Reply:
For a user without `change_submissions` permission, the button will be disabled (we tend to display UI elements as disabled rather than hiding, so users learn that there are more possibilities that they might "unlock" later). We check for `change_submissions` and not `delete_submissions` as deleting attachment will modify submission.

Reply:
After discussion on Zulip: fixed wrong permission being used in the modal